scripts/bot-deployment/DeployBotGCP.sh

```diff
@@ -1,33 +1,25 @@
 #!/bin/bash
 set -e
 
-if [ $# -ne 1 ]; then
-    echo "Incorrect number of arguments supplied! First and only argument is bot's name. From this the bot's config and service account will be inferred."
-    echo "example: ./DeployBotGCP.sh ethbtc-mainnet-monitor"
+if [ $# -lt 1 ]; then
+    echo "Incorrect number of arguments supplied! First (required) argument is the bot's name. From this the bot's config and service account will be inferred. Second (optional) argument is the GCP service account email to link with the bots."
```

Comment:
@mrice32 @chrismaree check this new logic for service account email here. First I check if user supplies one as the optional second CL-arg, otherwise it defaults to the one that starts with `emp-bot`

Reply:
this is a good solution I think. I like that you query the accounts then filter by the `emp-bot`. My only small nit is the way you've got it set up here irrespective if the user provides a service account or not they `$(gcloud iam service-accounts list --filter name:emp-bot --format 'value(email)')` logic is called. It would be ideal to do some kind of `if` `else` branching logic so it only pulls the service account iff there is no param provided. Other than this I'm happy with this solution now, thanks for refining it :)

Reply:
Makes sense.

Reply:
@chrismaree I added an `else` statement so we only query `gcloud iam service-accounts` if neccessary